app/controllers/application_controller.rb

```diff
@@ -5,6 +5,145 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "secret"
+  end
+
+  get '/' do 
+    erb :index
+  end
+
+  get '/signup' do 
```

Comment:
for lines 16- 137, I would separate this into 2 separate controllers: `users_controller.rb` and `tweets_controller.rb`. This will allow us to have separation of concerns, leaner controllers and more readable code.